src/scope-manager.js

```diff
@@ -446,6 +459,10 @@ var scopeManager = function(state, predefined, exported, declared) {
           "(type)" : type,
           "(token)": token,
           "(unused)": true };
+
+        if (!_current["(params)"]) {
+          _current["(params)"] = [];
```

Comment:
This seems like initialization logic. Does it really need to be dynamic? Since
  the `_newScope` function now receives the block type, is there any chance we
  can extend the logic there to create the array where necessary? My goal here
  is to make that function more descriptive of the "shape" of the scope object.
  As it stands, it's a little cagey about the `"(params)"` attribute (both its
  type and when it may be used). What do you think of:

```diff
diff --git a/src/jshint.js b/src/jshint.js
index 08dc777..187fa33 100644
--- a/src/jshint.js
+++ b/src/jshint.js
@@ -3872,7 +3872,7 @@ var JSHINT = (function() {
       advance("catch");
       advance("(");
 
-      state.funct["(scope)"].stack();
+      state.funct["(scope)"].stack("catchparams");
 
       if (checkPunctuators(state.tokens.next, ["[", "{"])) {
         var tokens = destructuringPattern();
diff --git a/src/scope-manager.js b/src/scope-manager.js
index e97a7e8..0494b84 100644
--- a/src/scope-manager.js
+++ b/src/scope-manager.js
@@ -24,7 +24,7 @@ var scopeManager = function(state, predefined, exported, declared) {
       "(breakLabels)": Object.create(null),
       "(parent)": _current,
       "(type)": type,
-      "(params)": null
+      "(params)": (type === "functionparams" || type === "catchparams") ? [] : null
     };
     _scopeStack.push(_current);
   }
@@ -254,7 +254,9 @@ var scopeManager = function(state, predefined, exported, declared) {
 
     /**
      * Tell the manager we are entering a new block of code
-     * @param {string} [type] - the type of the block. Valid values are "functionparams", "functionouter"
+     * @param {string} [type] - the type of the block. Valid values are
+     *                          "functionparams", "catchparams", and
+     *                          "functionouter"
      */
     stack: function(type) {
       var previousScope = _current;
@@ -465,9 +467,6 @@ var scopeManager = function(state, predefined, exported, declared) {
           "(token)": token,
           "(unused)": true };
 
-        if (!_current["(params)"]) {
-          _current["(params)"] = [];
-        }
         _current["(params)"].push(labelName);
       }
```